### PR TITLE
py-spyder* and dependencies: update to latest versions

### PR DESCRIPTION
**Pull request by reneeotten**
#### Description
- py-spyder: update to version 3.3.0, remove no longer supported Qt4 variant
- py-spyder-devel: update to de337a2 (20180706), remove no longer supported Qt4 variant

In addition, for these updates to work this PR includes:
- py-keyring: update to version 13.2.1, assume maintainership (new dependency)
- py-spyder-kernels: change to provide 0.x.x. branch needed for py-spyder
- py-spyder-kernels-devel: new port to provide 1.x.x. branch needed for py-spyder-devel

The changes in py-spyder-kernels* will inconvenience user that try to upgrade an currently installed  py-spyder-devel... It will try to install py-spyder-kernels-devel, but the now conflicting py-spyder-kernels is already installed. I don't know of a way to handle this automatically, the user will need to manually uninstall/deactivate py-spyder-kernels - if this is possibly, please let me know how to do so.

###### Type(s)

- [x] submission

###### Tested on

macOS 10.13.5 17F77
Xcode 9.4.1 9F2000
Python 2.7, 3.6

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [x] checked your Portfile with `port lint`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?

### CODE REVIEW

File: python/py-keyring/Portfile [hunk 1]
```diff
@@ -1,37 +1,46 @@
...
+master_sites        pypi:[string index ${python.rootname} 0]/${python.rootname}
+distname            keyring-${version}
 
-checksums           rmd160  6603d9bac6ff4b4947d2a87a6e9a2438eaf51641 \
-                    sha256  282132421bafc34b78a25ec07a78ce5539f3d358a290750fe79535fca2a693e0
+checksums           rmd160  09dd71fbfe911abd5f80f896c807d9fa6f420e20 \
+                    sha256  6364bb8c233f28538df4928576f4e051229e0451651073ab20b315488da16a58 \
+                    size    43140
 
 python.versions     27 34 35 36
```
Comment (pmetzger) [inline python/py-keyring/Portfile]: Will it support 3.7?

File: python/py-spyder-kernels-devel/Portfile [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+PortSystem          1.0
+PortGroup           python 1.0
+
+name                py-spyder-kernels-devel
+version             1.0.1
+categories-append   devel
+platforms           darwin
+license             MIT
+
+python.versions     27 35 36
```
Comment (pmetzger) [inline python/py-spyder-kernels-devel/Portfile]: Will it support 3.7?

File: python/py-spyder-kernels/Portfile [hunk 3]
```diff
@@ -4,12 +4,13 @@ PortSystem          1.0
...
 name                py-spyder-kernels
-version             1.0.0
+version             0.2.4
+epoch               1
 categories-append   devel
 platforms           darwin
 license             MIT
 
-python.versions     27 35 36
+python.versions     27 34 35 36
```
Comment (pmetzger) [inline python/py-spyder-kernels/Portfile]: Will it support 3.7?

**Comment by reneeotten** [comment]
@pmetzger it will probably support 3.7, it seems that upstream isn't testing for it yet though. Main issue is that their dependencies have not added py37 subports yet, so that will need to happen first.

**Comment by pmetzger** [comment]
Okay, please do update this when upstream says they can handle 3.7.

**Comment by pmetzger** [comment]
Merged. Thanks as always, @reneeotten!

**Comment by eborisch** [comment]
Sorry I haven't had time lately to chime in on these; thanks for carrying the ball, @reneeotten!

**Pull request merged**